glean-core/src/database/mod.rs

```diff
@@ -843,20 +878,51 @@ mod test {
                 _ => panic!("Unexpected data found"),
             }
         };
-
         db.iter_store_from(Lifetime::Ping, test_storage, None, &mut snapshotter);
         assert_eq!(1, found_metrics, "We only expect 1 Lifetime.Ping metric.");
 
-        // Make sure data was **not** persisted with rkv.
-        let store: SingleStore = unwrap_or!(
-            db.rkv
-                .open_single(Lifetime::Ping.as_str(), StoreOptions::create()),
-            panic!()
-        );
-        let reader = unwrap_or!(db.rkv.read(), panic!());
-        assert!(store
-            .get(&reader, &test_metric_id)
-            .unwrap_or(None)
-            .is_none());
+        {
+            // Make sure data **wasn't** persisted with rkv.
+            let store: SingleStore = db
+                .rkv
+                .open_single(Lifetime::Ping.as_str(), StoreOptions::create())
+                .unwrap();
+            let reader = db.rkv.read().unwrap();
+            assert!(store
+                .get(&reader, format!("{}#{}", test_storage, test_metric_id))
+                .unwrap_or(None)
+                .is_none());
+        }
+
+        // Attempt to persist data.
+        db.persist_ping_lifetime_data().unwrap();
+
+        {
+            // Make sure that data **was** persisted with rkv.
+            let store: SingleStore = db
+                .rkv
+                .open_single(Lifetime::Ping.as_str(), StoreOptions::create())
+                .unwrap();
+            let reader = db.rkv.read().unwrap();
+            assert!(store
+                .get(&reader, format!("{}#{}", test_storage, test_metric_id))
+                .unwrap_or(None)
+                .is_some());
+        }
+
+        // Verify that ping_lifetime_data was cleared and
```

Comment:
Mh, what are you testing here?

Reply:
I iterate over `Lifetime::Ping` stored data and make sure there is only one found metric, which would mean that we cleared the non-persisted data, since I just checked that data is indeed saved to rkv above.

That is not the most intuitive test, let me know if you have better ideas :)

Reply:
What about this: we leave the existing test alone, then we can add another test that does the following:

- records a metric with `lifetime: ping`
- calls the persist function
- records a different metric with `lifetime: ping`
- checks that the first metric is in rkv and the second is in memory
- calls the persist function again
- checks that both metrics are in rkv and none is in memory

What do you think?

Reply:
I did this. I think it is clearer although it is a very long test :)